WordPress/gutenberg
# Override all the labels of the pattern categories taxonomy

closes #57076 

## What?

The site editor has been updated to use the normal post taxonomies component, this highlighted the fact that the "pattern categories" taxonomy didn't have all the labels defined properly. This PR fixes that.

## Testing Instructions

 - Open a synced pattern in the site editor
 - Notice the "pattern categories" panel on the editor sidebar
 - It says "Add pattern category" rather than "Add tag"
